test/forms/CancelDeclarationSpec.scala

```diff
@@ -0,0 +1,96 @@
+      val cancellationDeclaration = correctCancelDeclaration
+      val metadata = cancellationDeclaration.createCancellationMetadata("N4tur380y")
+      metadata.declaration must be(defined)
+      metadata.declaration.get.functionCode must be(defined)
+      metadata.declaration.get.functionCode.get must be(13)
+      metadata.declaration.get.functionalReferenceId must be(defined)
+      metadata.declaration.get.functionalReferenceId.get must be("1SA123456789012-1FSA1234567")
+      metadata.declaration.get.id must be(defined)
+      metadata.declaration.get.id.get must be("87654321")
+      metadata.declaration.get.typeCode must be(defined)
+      metadata.declaration.get.typeCode.get must be("INV")
+      metadata.declaration.get.submitter must be(defined)
+      metadata.declaration.get.submitter.get.id must be(defined)
+      metadata.declaration.get.submitter.get.id.get must be("N4tur380y")
+      metadata.declaration.get.additionalInformations.size must be(1)
+      metadata.declaration.get.additionalInformations.head.statementDescription must be(defined)
+      metadata.declaration.get.additionalInformations.head.statementDescription.get must be("Some description")
+      metadata.declaration.get.additionalInformations.head.statementTypeCode must be(defined)
+      metadata.declaration.get.additionalInformations.head.statementTypeCode.get must be("AES")
+      metadata.declaration.get.additionalInformations.head.pointers.size must be(2)
+      metadata.declaration.get.additionalInformations.head.pointers.head.sequenceNumeric must be(defined)
+      metadata.declaration.get.additionalInformations.head.pointers.head.sequenceNumeric.get must be(1)
+      metadata.declaration.get.additionalInformations.head.pointers.head.documentSectionCode must be(defined)
+      metadata.declaration.get.additionalInformations.head.pointers.head.documentSectionCode.get must be("42A")
+      metadata.declaration.get.additionalInformations.head.pointers(1).documentSectionCode must be(defined)
+      metadata.declaration.get.additionalInformations.head.pointers(1).documentSectionCode.get must be("06A")
+      metadata.declaration.get.amendments.size must be(1)
+      metadata.declaration.get.amendments.head.changeReasonCode must be(defined)
+      metadata.declaration.get.amendments.head.changeReasonCode.get must be(NoLongerRequired.toString)
+    }
+  }
+
+}
+
+object CancelDeclarationSpec {
+  val correctCancelDeclaration =
+    CancelDeclaration(
+      functionalReferenceId = "1SA123456789012-1FSA1234567",
+      declarationId = "87654321",
+      statementDescription = "Some description",
+      changeReason = NoLongerRequired.toString
+    )
+
+  val emptyRepresentativeDetails =
+    CancelDeclaration(
+      functionalReferenceId = "",
+      declarationId = "",
+      statementDescription = "",
+      changeReason = NoLongerRequired.toString
+    )
+
+  val correctCancelDeclarationJSON: JsValue = JsObject(
```

Comment:
I'm just thinking that to prepare a JsValue is not enought to use `Json.toJson(correctCancelDeclaration)` but I don't know where you need this etc, so it's just a comment

Reply:
That is a very good point. This was a pattern we have in our code but it would make more sense to use that.

Reply:
I just did that and it worked!!!! :) Well that was very useful